samples/rpc-priority-queryPartitions.js

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// sample-metadata:
+//  title: Create and Execute Query partitions with RPC priority
+//  usage: node rpc-priority-queryPartitions.js <INSTANCE_ID> <DATABASE_ID> <PROJECT_ID>
+
+'use strict';
+
+async function main(instanceId, databaseId, projectId) {
+  // [START spanner_rpc_priority_query_partitions]
+  // Imports the Google Cloud client library.
+  const {Spanner, protos} = require('@google-cloud/spanner');
+  const Priority = protos.google.spanner.v1.RequestOptions.Priority;
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  async function createAndExecuteQueryPartitionsWithRpcPriority(
+    instanceId,
+    databaseId
+  ) {
+    // Gets a reference to a Cloud Spanner instance and database
+    const instance = spanner.instance(instanceId);
+    const database = instance.database(databaseId);
+    const [transaction] = await database.createBatchTransaction();
+
+    const [partitions] = await transaction.createQueryPartitions({
+      sql: 'SELECT * FROM Singers',
+      requestOptions: {
```

Comment:
I think it would be better for the sample to add the priority to the `transaction.execute(partition)` calls below (if those support an RPC priority configuration), as this call is not a call that is going to take a long time. This could also confuse users into thinking that setting a priority here would have an effect for all the partitions that are returned.

Reply:
Adding priority to ```transaction.execute(partition)``` is not supported. Each partition query will inherit the options set from the partitionQuery call.

Reply:
Hmm... I think that is only partly correct. It seems that using a `partition` directly for the same client will indeed inherit the priority from the `createQueryPartitions` call. But the purpose of the batch read API is to also distribute the read operation to multiple clients by sending the partition IDs to other clients, and those clients would need to specify the priority, as the priority is not something that is 'remembered' server side for a partition (or am I missing something here?).

But I think that you can include query options when you are calling `transaction.execute(partition)`. That is, from what I can see, is happening in the implementation of `createQueryPartitions`. It will just add the query options that were given for the `createQueryPartitions` to the resulting partitions that are being returned: https://github.com/googleapis/nodejs-spanner/blob/2839d2317ca7368d288ee9d7feb806f0ac2069c6/src/batch-transaction.ts#L173

Reply:
Even if partition is assigned to a different client , the partition will still be having priority assigned with it. Right ?
Added a comment that priority can be changed for a partition.

Reply:
Added the priority in execute partition call.